### PR TITLE
fix `make install`

### PR DESCRIPTION
**Pull request by andlaus**
some CMakeLists.txt needed some adaptation after the unit
changes. sorry for not noticing this earlier...

### CODE REVIEW

**Comment by jokva** [comment]
I've also added these fixes to my shared_ptr purge, so if desired this can be fixed as a part of that.

**Comment by andlaus** [comment]
hm: the appveyor errors seem to be caused by some GCCisms in Units.hpp: as far as I understand the error message, it does not seem to swallow the `__attribute__((unused))` statement which was added to squelch some "unused variable" GCC warnings. If I remember correctly, that one has not been required anymore after the units have been marked as `constexpr`? I pushed a patch, let's see if it helps...

> I've also added these fixes to my shared_ptr purge, so if desired this can be fixed as a part of that.

I'm fine with this. do as you prefer.

**Comment by jokva** [comment]
> hm: the appveyor errors seem to be caused by some GCCisms in Units.hpp: as far as I understand the error message, it does not seem to swallow the __attribute__((unused)) statement which was added to squelch some "unused variable" GCC warnings. If I remember correctly, that one has not been required anymore after the units have been marked as constexpr? I pushed a patch, let's see if it helps...

Also fixed in my shared_ptr purge.

File: opm/parser/eclipse/Units/Units.hpp [hunk 1]
```diff
@@ -242,15 +242,6 @@ namespace Opm {
                 return q / unit;
             }
         } // namespace convert
-
-
-#ifndef HAS_ATTRIBUTE_UNUSED
```
Comment (akva2) [inline opm/parser/eclipse/Units/Units.hpp]: yes. @babrodtk please confirm, even though it doesn't matter any longer this block can definitely go.

**Comment by akva2** [comment]
yeah, msvc does not support the attribute unused. this bit us now because this code hasn't been built on msvc before and the ifdef was wrong as you pointed out triggering the issue.

**Comment by andlaus** [comment]
hm, depending on how long #939 takes until it stabilizes, we should consider merging this PR before?

**Comment by atgeirr** [comment]
I'd like to have this merged if possible, to ensure Jenkins tests work properly. Is that OK?

File: opm/parser/eclipse/Units/Units.hpp (hunk 1 above)
Comment (babrodtk) [inline opm/parser/eclipse/Units/Units.hpp]: @akva2, @andlaus I agree that it should be #if

**Comment by andlaus** [comment]
> Is that OK?

I suppose it is. @jokva will probably get some (very minor AFAICS) conflicts with this shared_ptr purge, though...

**Comment by pgdr** [comment]
@atgeirr @andlaus no, let's wait for #939.

**Comment by andlaus** [comment]
how long is this gonna take approximately?

**Comment by jokva** [comment]
Probably today (finally got bloody mpi+zoltan to compile).

**Comment by atgeirr** [comment]
This is *really* holding us (well, Jenkins) back right now, and I do not see the conflict with #939 at all (other than perhaps the same fix being applied there). That is a quick rebase away with no conflicts (or it should be, if not please tell me why).

Semi-related: I have asked @GitPaean to try to fix the opm-grid Zoltan issues with this PR so that we can get #939 in as quickly as possible. He has the same setup as @blattms.

**Comment by atgeirr** [comment]
> Probably today (finally got bloody mpi+zoltan to compile).

Saw this after I posted my comment above: does this mean that you have already fixed opm-grid, so all we have to wait for is the final Jenkins run?

**Comment by jokva** [comment]
> Saw this after I posted my comment above: does this mean that you have already fixed opm-grid, so all we have to wait for is the final Jenkins run?

Seems so, but you never know.

**Comment by atgeirr** [comment]
> Seems so, but you never know.

Based on that, I am willing to wait.

**Comment by blattms** [comment]
> @atgeirr @andlaus no, let's wait for #939.

I totally disagree with this approach as I consider it very bad style
to collect multiple independent fixes in PRs that change the
semantics. Every self contained fix deserves its own PR. This makes it
a lot easier to identify the origin of problems (bugs, performance
regressions, etc.) and might save a lot of time afterwards.

There please merge this independently of the shared_ptr stuff.

**Comment by jokva** [comment]
A PR doesn't really help with that, although commits do. The fix is in its own commit (and a better one in my opinion, although this can certainly be changed to be just that).

It will take just as long though considering this doesn't have a confirmed jenkins build. I can also rip out my own fix and submit it separately, but again, no we probably still have to wait for jenkins.

**Comment by pgdr** [comment]
>>  @atgeirr @andlaus no, let's wait for #939.

> I totally disagree [...] Every self contained fix deserves its own PR. This makes it
a lot easier to identify the origin of problems [...]

We don't use PRs to identify the origin of problems, but commits.

> There please merge this independently of the shared_ptr stuff.

Please do not.  Please understand that merging the "shared_ptr stuff" is a big job that spans every repository.  Every single new merge puts the "shared_ptr stuff" back many hours.  We do not merge until Jenkins or Travis gives the green light.

**Comment by atgeirr** [comment]
> We don't use PRs to identify the origin of problems, but commits.

In practice, yes we do. Commits are often too small a unit to use in discussions about a problem.

> Please do not. Please understand that merging the "shared_ptr stuff" is a big job that spans every repository. Every single new merge puts the "shared_ptr stuff" back many hours. We do not merge until Jenkins or Travis gives the green light.

OK. But the only reason this is a problem was that #939 had a fix for the same problem added to it. Otherwise we could have merged this without causing any problem for #939.

**Pull request closed without merging**